modules/RsIskraMeter/src/main.rs

```diff
@@ -0,0 +1,1197 @@
+                None => anyhow::bail!("Received None as value"),
+                Some(inner) => {
+                    if inner.len() != N {
+                        anyhow::bail!("Expected size {}, received size {N}", inner.len())
+                    }
+                    let mut res = [0; N];
+                    for (ss, dd) in std::iter::zip(inner, &mut res) {
+                        *dd = ss as u16;
+                    }
+                    Ok(res)
+                }
+            },
+            StatusCodeEnum::Error => anyhow::bail!("StatusCodeEnum::Error"),
+        }
+    }
+}
+
+/// Custom extension of the auto generated `TransactionStartResponse`.
+impl TransactionStartResponse {
+    /// Constructs an error response from the input.
+    /// * `error`: The error type.
+    fn from_err<E>(error: &E) -> Self
+    where
+        E: Debug,
+    {
+        Self {
+            error: Some(format!("{error:?}")),
+            signed_meter_value: None,
+            status: TransactionRequestStatus::UNEXPECTED_ERROR,
+            transaction_max_stop_time: None,
+            transaction_min_stop_time: None,
+        }
+    }
+}
+
+/// Custom extension of the auto generated `TransactionStopResponse`.
+impl TransactionStopResponse {
+    /// Constructs an error response from the input.
+    /// * `error`: The error type.
+    fn from_err<E>(error: &E) -> Self
+    where
+        E: Debug,
+    {
+        Self {
+            error: Some(format!("{error:?}")),
+            signed_meter_value: None,
+            status: TransactionRequestStatus::UNEXPECTED_ERROR,
+        }
+    }
+}
+
+#[derive(everest_serde::Serialize, Clone)]
+#[serde(crate = "everest_serde")]
+/// The serialization according to the Open charge metering format. See
+/// https://github.com/SAFE-eV/OCMF-Open-Charge-Metering-Format/blob/master/OCMF-en.md
+/// for details.
+///
+/// The struct does not implement everything and also our config does not
+/// implement everything. Extend it as needed.
+struct OcmfData {
+    #[serde(rename = "FV")]
+    /// Version of the data format in the representation.
+    format_version: String,
+
+    #[serde(rename = "GI")]
+    /// Identifier of the manufacturer for the system which has generated the
+    /// present data
+    gateway_identification: String,
+
+    #[serde(rename = "GS")]
+    /// Serial number of the above mentioned system.
+    gateway_serial: String,
+
+    #[serde(rename = "GV")]
+    /// Version designation of the manufacturer for the software.
+    gateway_version: String,
+
+    #[serde(rename = "PG")]
+    /// Pagination of the entire data set, i.e. the data that is combined in one
+    /// signature.
+    pagination: String,
+
+    #[serde(rename = "MV")]
+    /// Manufacturer identification of the meter, name of the manufacturer.
+    meter_vendor: String,
+
+    #[serde(rename = "MM")]
+    /// Model identification of the meter.
+    meter_model: String,
```

Comment:
we should probably also meter_model and meter_serial (we read them) and also meter_vendor (iskra?) and probably also meter_firmware (read under register 30013) - opinions @golovasteek @hikinggrass ?

Reply:
we should what?

Reply:
we should not forget the verb in a sentence